File: src/visual-graph/builder/if-test-node-id.test.ts
import { describe, expect, test } from "vitest";

import { ifTestNodeId } from "./if-test-node-id.js";

describe("ifTestNodeId", () => {
  test.each([
    { parentScopeId: "scope1", offset: 42, expected: "if_test_scope1_42" },
    { parentScopeId: "a.b", offset: 100, expected: "if_test_a_b_100" },
    { parentScopeId: "s", offset: 0, expected: "if_test_s_0" },
    { parentScopeId: "", offset: 7, expected: "if_test__7" },
  ])(
    "ifTestNodeId($parentScopeId, $offset) = $expected",
    ({ parentScopeId, offset, expected }) => {
      expect(ifTestNodeId(parentScopeId, offset)).toBe(expected);
    },
  );
});

File: crates/unsnarl-visual-graph/src/builder/if_test_node_id_test.rs
//! Sibling tests for [`if_test_node_id`].

use super::if_test_node_id;

#[test]
fn alphanumeric_scope_id_pass_through() {
    assert_eq!(if_test_node_id("scope1", 42), "if_test_scope1_42");
}

#[test]
fn specials_in_scope_id_are_replaced() {
    assert_eq!(if_test_node_id("a.b", 100), "if_test_a_b_100");
}

#[test]
fn zero_offset_renders_as_zero() {
    assert_eq!(if_test_node_id("s", 0), "if_test_s_0");
}

#[test]
fn empty_scope_id_keeps_separator() {
    assert_eq!(if_test_node_id("", 7), "if_test__7");
}
